clustermesh/clustermesh.go

```diff
@@ -672,6 +674,27 @@ func getDeprecatedName(secretName string) string {
 	}
 }
 
+// getDeprecatedSecret attempts to retrieve a secret using one or more deprecated names
+// There are now multiple "layers" of deprecated secret names, so we call this function recursively if needed
+func (k *K8sClusterMesh) getDeprecatedSecret(ctx context.Context, client k8sClusterMeshImplementation, secretName string, defaultName string) (*corev1.Secret, error) {
+
+	deprecatedSecretName := getDeprecatedName(secretName)
+	if deprecatedSecretName == "" {
+		return nil, fmt.Errorf("unable to get secret %q and no deprecated names to try", secretName)
+	}
+
+	k.Log("Trying to get secret %s by deprecated name %s", secretName, deprecatedSecretName)
+
+	secret, err := client.GetSecret(ctx, k.params.Namespace, deprecatedSecretName, metav1.GetOptions{})
+	if err != nil {
```

Comment:
Should we check the error kind here? Currently, we don't distinguish missing secrets and other errors.

Reply:
There are many instances in this file of the error type not being checked. I think that cleanup would be best left to another PR.

In this case among those in this file, I think it's least critical to check for errors, because Kubernetes internal APIs are relatively unstable, so if there is an error type that we fail to check for which has a similar meaning to "not found" (e.g. malformed content of some type, which could indicate that we _should_ check for a deprecated name), then we should at least try to request the deprecated name.